src/Abp/Runtime/Caching/AbpCacheBase.cs

```diff
@@ -258,7 +258,35 @@ public virtual async Task<TValue[]> GetAsync(TKey[] keys, Func<TKey, Task<TValue
             return items;
         }
 
-        public abstract TValue GetOrDefault(TKey key);
+        protected abstract bool TryGetValue(TKey key, out TValue value);
+
+        protected virtual KeyValuePair<bool, TValue>[] TryGetValues(TKey[] keys)
+        {
+            var pairs = new List<KeyValuePair<bool, TValue>>();
+            foreach (var key in keys)
+            {
+                var found = TryGetValue(key, out TValue value);
+                pairs.Add(new KeyValuePair<bool, TValue>(found, value));
+            }
+            return pairs.ToArray();
+        }
+
+        protected virtual Task<KeyValuePair<bool, TValue>> TryGetValueAsync(TKey key)
```

Comment:
1. `KeyValuePair<bool, TValue>` does not seem like an appropriate return type. How about our own `ConditionalValue<TValue>` struct similar to the return type of Azure's [`IReliableDictionary<TKey,TValue>.TryGetValueAsync`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicefabric.data.collections.ireliabledictionary-2.trygetvalueasync?view=azure-dotnet) method?

2. `GetOrDefaultAsync(TKey key)` should be changed to call this instead of doing `Task.FromResult(GetOrDefault(key))`.

Similarly for the multi-key counterparts.

Reply:
Sounds good to have `ConditionalValue<TValue>`

Reply:
@acjh done 😄